ReactiveCocoa/RACSignal+Operations.m

```diff
@@ -560,7 +578,7 @@ - (RACSignal *)flatten:(NSUInteger)maxConcurrent {
 			NSCAssert([signal isKindOfClass:RACSignal.class], @"Expected a RACSignal, got %@", signal);
 
 			@synchronized (subscriber) {
-				if (maxConcurrent > 0 && activeDisposables.count >= maxConcurrent) {
+				if (maxConcurrent > 0 && (activeDisposables.count >= maxConcurrent || pendingSubscription)) {
```

Comment:
I don't think this works generally. It works for `-concat` because `maxConcurrent` is `1`, but `flatten:` needs to handle `n`—including `n` pending subscriptions.

Think of the case where a signal is `flatten:10`. If 3 threads race to send a signal through the `flatten:`, all 3 should be immediately subscribed to—even if one of them is `pendingSubscription`.

So I think `pendingSubscription` should be an `NSUInteger` of pending subscriptions, and this should check `maxConcurrent > 0 || (activeDisposables.count + pendingSubscriptions >= maxConcurrent)`. Or maybe it would be simpler to add a `activeCount` and use/manipulate that instead.

This is probably relatively uncommon—`flatten:` didn't seem to be widely used. But it could lead to some bad behavior.